goss-testing/scripts/data-validator.py

```diff
@@ -192,103 +206,98 @@ def are_hosts_sane(data, desired_keys):
 
 
 def validate_ntp(data):
+  """Check ntp data.
+  """
+
   err = 0
   if are_valid_ip_masks(data, ['ntp', 'allow']): err = 1
   if are_hosts_sane(data, ['ntp', 'peers']): err = 1
   if are_hosts_sane(data, ['ntp', 'servers']): err = 1
   return err
 
 
-def score_params(node_class):
-  err = 0
-
-  lengths = []
-  counts = []
-  for blob in node_class:
-    lengths.append(blob['params-length'])
-  
-  for i in lengths:
-    occ = lengths.count(i)
-    counts.append(occ)
-    for blob in node_class:
-      if i == blob['params-length']:
-        blob['occurrence'] = occ
-
-  baseline = max(counts)
-
-  searches = [ "bond=bond0:mgmt0,mgmt1",
-               "ifname=mgmt0",
-               "ip=mgmt0",
-               "ifname=mgmt1",
-               "ip=mgmt1",
-               "hostname=" ]
-
-  all_param_strings = []
-  for blob in node_class:
-    for k,v in blob.items():
-      if k == "params-stripd":
-        all_param_strings.append(v)
-    if blob['occurrence'] == baseline:
-      for search in searches:
-        if re.search(search, blob['params-orig']):
-          blob['valid'] = True
-        else:
-          blob['valid'] = False
-    else:
-      blob['valid'] = False
-
-  unique_param_strings = set(all_param_strings)
-  if len(unique_param_strings) > 1:
-    for blob in node_class:
-      for params in unique_param_strings:
-        if params == blob['params-stripd']:
-          blob['valid'] = False
-
-
-  for blob in node_class:
-    if blob['valid'] == False:
-      print_err(f"WARN: {blob['hostname']}: has boot params that differ from at least one other node\n{blob['params-orig']}\n")
-      err = 1
-
-  print_err("------------------------------------------------------------")
-
-  if err == 1:
-    return err
-
-
 def boot_params(data):
-  """Attempt to validate boot parameters from BSS
+  """Confirm that params required for booting are
+     present.
   """
-  worker = []
-  storage = []
-  management = []
-  err = 0
 
-  macreg = re.compile(r':([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}:([0-9a-fA-F]){2}')
-  hostreg = re.compile(r'hostname=[a-zA-Z]*\-[a-zA-Z0-9]*')
-  httpreg = re.compile(r's=http://.*/')
+  err = 0
+  ncn_params = [
+	"biosdevname=1",
+	"console=tty0",
+	"console=ttyS0,115200",
+	"crashkernel=[0-9]*M",
+	"ds=nocloud-net;s=http://",
+	"hostname=",
+	"ifname=mgmt0",
+	"ifname=mgmt1",
+	"initrd=initrd.img.xz",
+	"iommu=pt",
+	"log_buf_len=1",
+	"metal.no-wipe=",
+	"metal.no-wipe=1",
```

Comment:
Why is this listed twice?

I'd actually advocate for not checking for `metal.no-wipe`. There's too many permutations:
- fresh install will have `metal.no-wipe=0`
- a finished install will have `metal.no-wipe=1`
- a disk boot always has `metal.no-wipe=1`
- a rebuild of any NCN uses `metal.no-wipe=0`
- an upgrade of a k8s-master and k8s-worker uses `metal.no-wipe=0`
- an upgrade of a storage-ceph node uses `metal.no-wipe=1`

Maybe ignoring this one would be easier?

Reply:
Plus we have other tests to look for this already, I think

Reply:
Pretty sure upgrades validate this, or at least pave over and make it the desired value.

Reply:
It shouldn't be listed twice. I'll remove it entirely.